src/components/memoBlock/memo.component.tsx

```diff
@@ -30,64 +41,108 @@ function MemoBlock(props: Props): JSX.Element {
-    shuffledMemoBlockDuplicated.splice(memoBlock.index, 1, flippedBlock);
-    setShuffledMemoBlocks(shuffledMemoBlockDuplicated);
+  function handleClick(memoBlock: MemoBlock) {
+    setShuffledMemoBlocks(
+      getSetWithMemoBlock(shuffledMemoBlocks, asFlipped(memoBlock))
+    );
 
-    if (selectedBlock === null || !selectedBlock) {
+    if (noneSelection()) {
       setSelectedBlock(memoBlock);
-    } else if (selectedBlock && selectedBlock.image === memoBlock.image) {
+    } else if (equalSelection()) {
+      // Pair selected, no need to keep tracking on it anymore.
       setSelectedBlock(null);
     } else {
+      flipAndResetSelection();
+    }
+
+    function getSetWithMemoBlock(memoBlocks: MemoBlocks, memoBlock: MemoBlock) {
+      memoBlocks = cloneDeep(memoBlocks);
+      memoBlocks.splice(memoBlock.index, 1, memoBlock);
+
+      return memoBlocks;
+    }
+
+    function asFlipped(memoBlock: MemoBlock) {
+      memoBlock = cloneDeep(memoBlock);
+      memoBlock.flipped = true;
+
+      return memoBlock;
+    }
+
+    function asNotFlipped(memoBlock: MemoBlock) {
+      memoBlock = asFlipped(memoBlock);
+      memoBlock.flipped = false;
+
+      return memoBlock;
+    }
+
+    function noneSelection() {
+      return selectedBlock === null || !selectedBlock;
+    }
+
+    function equalSelection() {
+      return selectedBlock && selectedBlock.image.id === memoBlock.image.id;
+    }
+
+    function flipAndResetSelection() {
       setAnimating(true);
 
       setTimeout(() => {
         if (selectedBlock) {
-          shuffledMemoBlockDuplicated.splice(memoBlock.index, 1, memoBlock);
-          shuffledMemoBlockDuplicated.splice(
-            selectedBlock.index,
-            1,
-            selectedBlock
+          let memoBlocksWithSelectionFlipped = getSetWithMemoBlock(
+            shuffledMemoBlocks,
+            asNotFlipped(memoBlock)
+          );
+          memoBlocksWithSelectionFlipped = getSetWithMemoBlock(
+            memoBlocksWithSelectionFlipped,
+            asNotFlipped(selectedBlock)
           );
-          setShuffledMemoBlocks(shuffledMemoBlockDuplicated);
+
+          setShuffledMemoBlocks(memoBlocksWithSelectionFlipped);
           setSelectedBlock(null);
           setAnimating(false);
         }
       }, 1000);
     }
   }
+
+  const handleSelectionLevels = (level: string) => {
+    const [row, col] = level.split("*");
+    const levels = {
+      row: parseInt(row),
+      col: parseInt(col),
+    };
+    setSelectedLevel(levels);
+  };
+
+  const gridTemplateColumns = `repeat(${selectedLevel.col}, 1fr)`;
+  const gridTemplateRows = `repeat(${selectedLevel.row}, 1fr)`;
+
   return (
-    <>
+    <div className={props.className}>
       <Title>MemoBlocks Game!</Title>
-      <div className={`${props.className} content`}>
+      <h2 className="levels-title"> Select level:</h2>
+      <RadioButtons levels={levels} onClick={handleSelectionLevels} />
+
+      <div
+        className={"content"}
```

Comment:
className="content"

No need for curly abraces as it is getting an escalar value.